ajax/rack.php

```
@@ -42,8 +45,7 @@
 Session::checkLoginUser();
 
 if (!Session::haveRight('datacenter', UPDATE)) {
-    http_response_code(403);
-    die;
+    throw new AccessDeniedHttpException();
 }
 if (!isset($_REQUEST['action'])) {
     exit();
```

You can throw a BadRequest exception here.